### PR TITLE
Add more fields to Asset editing modal

### PR DESCRIPTION
**Pull request by aadgrant**
Firstly, brilliant plugin, a pleasure to use and edit!

Key features:
- Added new fields to Asset editing modal
  - Title (defaults to original filename)
  - Location (set a geopoint Latitude, Longitude, Altitude)
  - Attribution
- Asset editing modal now has overflow set
- All fields can be toggled on/off in the config!

image

TODO: Add a custom metadata field for projects?

Happy to talk about development of this further!

### CODE REVIEW

**Comment by Aratramba** [comment]
Thanks for contributing @aadgrant, we'll look into it.

**Comment by aadgrant** [comment]
Thanks, I've made one last update - you can now define custom textarea, text, number, checkbox fields.

I've stored these in an attributes object to avoid image assets getting too messy.

Nothing else planned but I'm using this in three projects so we'll see what comes up!

**Comment by dennispassway** [comment]
Thanks so much for your contribution!

The PR looks good but I think it involves too much. Your contribution started a discussion with us internally where we discussed what fields belong in the media library.

We concluded that an important concept in the plugin is:
If it's app data, it should be in your schemas and if it's data about the media then it should be in the media library.

Another thing that's important is simplicity. That why we have the tags, because (almost) everything can be put in a tag so you can filter on it in the media library.
I think that adding multiple fields makes the plugin more complex and less user friendly. At the moment I believe that most of it's users would be helped more with a simpler approach, thus less fields.

Could you tell a little more about your use case? Where do you need the `premiumPhoto` or `location` attribute for example?
You don't seem to be use it in the media library plugin.

For this use case I would suggest to create a schema that references the image:
```typescript
export const customImageSchema = {
  name: "customImageSchema",
  type: "object",
  title: "Image",
  fields: [
    {
      name: "image",
      type: "image", // This uses the media library, and therefore this schema extends the asset object.
    },
    {
      name: "location",
      type: "location",
    },
    {
      name: "premium",
      type: "boolean",
    },
  ],
};
```

Having said that: I like the idea of turning fields on and off in the config.
That sparked the idea that we could introduce some hooks in the future that allow you to add to the data of the asset, without it impacting the UI.

In this pull request you have convinced me that a little more data would be helpful.
I think 2 fields could be added: `attribution` and `title` (that defaults to filename).

Since `description` and `caption` are usually context specific and `location` is a rare use case, I would solve these in your schemas.

Could you split the pull request, so that we can merge the attribution and title field and discuss the field configs a little more in another issue?

I look forward to hearing your views on this.

**Comment by coreyward** [comment]
I like the approach @aadgrant is using here. I always want to keep image data with the image, and that includes things like the location, default alt text, and source/attribution data (a la Unsplash source plugin). I don't think something like the location data or attribution should come out of the box, but a handy example configuration for adding them as custom fields via the config seems like a great approach.

File: src/components/AssetModal.tsx [hunk 1]
```diff
@@ -77,7 +88,7 @@ const StyledInfoContainer = styled.div`
   font-size: 14px;
   font-weight: 400;
   line-height: 1.4;
-
+˝
```
Comment (coreyward) [inline src/components/AssetModal.tsx:91]: Looks like this might be a typo?

Comment (aadgrant) [reply src/components/AssetModal.tsx]: Thanks!

**Comment by aadgrant** [comment]
Thanks for the comments.

To provide some use cases I currently have open three projects where this functionality is important:
- a travel website, showing photographs on a map
- a museum website which shows the changing landscape of Yorkshire through a map of historical photographs
- a travel photography gallery, again showing photos on a map

Currently, for the museum, I am using a similar schema method to the one mentioned above. It's the most intuitive solution at the moment but its also cumbersome for the user: for every image a new article must be created, the photo selected from the library or uploaded. Partly, we are limited by Sanity's interface (ie, where the edit button appears).

In these cases, the location of the photo is always the same (where it was taken or what it is a photo of). I access it wherever I use the photo to provide cross links:
- The map page: used to place the image on the map
- The media browser: used to visit map and zoom to location
- Locations page: used to show nearby photos by their location

For the gallery, whether the photo can be bought or not is relative to the photo, not to where it is used on the site - for example, if the image was also featured on the gallery's blog I might want to link to a purchase button.

I very much understand your concerns over simplicity, it was an obstacle even in making these changes. A customisable interface seems the best way forward. Thinking about whether its app date or media data, I would argue most of this data is related to the media itself.

**Comment by Aratramba** [comment]
You make good points @aadgrant and @coreyward - we've reopened the discussion 👍

**Comment by dennispassway** [comment]
You have convinced me for the use of custom fields @aadgrant . I'm working on a PR on your PR so we can discuss some concept from there.

**Comment by dennispassway** [comment]
I have reviewed your branch and made some changes. I made the fields configurable by making the config expandable. I think this is a more scalable option than turning on and off some fields. I would like to hear your opinion on this @aadgrant so we can resolve this issue :) The PR: https://github.com/dennispassway/sanity-plugin-media-library/pull/11

**Pull request merged**